integration-tests/utils/integration-helpers.js

```diff
@@ -117,34 +117,33 @@ const withBilling = test => async options => {
 
 const trimClustersOldestFirst = _.flow(
   _.remove({ status: 'Deleting' }),
-  _.remove({ status: 'Creating' }),
   _.sortBy('createdDate')
 )
 
 const currentCluster = _.flow(trimClustersOldestFirst, _.last)
 
-const getCurrentCluster = withUserToken(async ({ billingProject, context, testUrl, token }) => {
+const getCurrentCluster = withUserToken(async ({ billingProject, context, email, testUrl, token }) => {
   const ajaxPage = await context.newPage()
   await ajaxPage.goto(testUrl)
   await signIntoTerra(ajaxPage, token)
 
-  const clusters = await ajaxPage.evaluate(billingProject => {
-    return window.Ajax().Clusters.list({ googleProject: billingProject })
-  }, billingProject)
+  const clusters = await ajaxPage.evaluate((billingProject, email) => {
+    return window.Ajax().Clusters.list({ googleProject: billingProject, creator: email })
+  }, billingProject, email)
 
   await ajaxPage.close()
   return currentCluster(clusters)
 })
 
-const deleteCluster = withUserToken(async ({ billingProject, context, testUrl, token }) => {
+const deleteCluster = withUserToken(async ({ billingProject, context, email, testUrl, token }) => {
   const ajaxPage = await context.newPage()
   await ajaxPage.goto(testUrl)
   await signIntoTerra(ajaxPage, token)
 
-  const currentC = await getCurrentCluster({ billingProject, context, testUrl })
-  currentC && await ajaxPage.evaluate((currentC, billingProject) => {
-    return window.Ajax().Clusters.cluster(billingProject, currentC.runtimeName).delete()
-  }, currentC, billingProject)
+  const currentC = await getCurrentCluster({ billingProject, context, email, testUrl })
+  currentC && await ajaxPage.evaluate(currentC => {
+    return window.Ajax().Clusters.cluster(currentC.googleProject, currentC.runtimeName).delete()
```

Comment:
TOL: Since this is effectively a cleanup operation, I wonder if it would be better to find and delete ALL clusters, rather than just the current one. We only expect there to be one right now, but we could make it resilient to 0 or N to cover edge cases and/or future changes to the test code.

Reply:
We could do that, as long as we only delete runtimes for the current user.

A happy side effect would be eliminating the duplicate implementations of `trimClustersOldestFirst`. We probably still want to filter out runtimes that are already `Deleting`, but that makes logical sense and wouldn't be a duplication of intent with the UI code.

I'll sketch this out and either push another commit or post a follow-up comment here.